File: web/lib/branding.ts
/**
 * Central product identity for the web app.
 *
 * Mirrors `qguide/app/branding.py`. Every visible mention of the product name,
 * tagline, support address or legal entity must read from here — renaming the
 * product is then one edit (or a set of env vars), not a codebase search.
 *
 * Values can be overridden at build time with NEXT_PUBLIC_* env vars, and the
 * backend's /branding endpoint is the runtime source of truth if you prefer to
 * drive it from the server (see `useBranding`).
 */

const env = (key: string, fallback: string) => {
  const v = process.env[key as keyof NodeJS.ProcessEnv] as string | undefined;
  return v && v.trim() ? v.trim() : fallback;
};

export interface BrandingConfig {
  APP_NAME: string;
  APP_SHORT_NAME: string;
  APP_DESCRIPTION: string;
  TAGLINE: string;
  LOGO_MARK: string;
  LOGO_URL: string;
  SUPPORT_EMAIL: string;
  LEGAL_COMPANY_NAME: string;
  WEBSITE_URL: string;
  DOCS_URL: string;
}

export const BRANDING: BrandingConfig = {
  APP_NAME: env("NEXT_PUBLIC_APP_NAME", "QGuide"),
  APP_SHORT_NAME: env("NEXT_PUBLIC_APP_SHORT_NAME", env("NEXT_PUBLIC_APP_NAME", "QGuide")),
  APP_DESCRIPTION: env(
    "NEXT_PUBLIC_APP_DESCRIPTION",
    "Context-aware, explainable CRISPR guide RNA design and analysis."
  ),
  TAGLINE: env("NEXT_PUBLIC_APP_TAGLINE", "Explainable, outcome-first CRISPR guide design"),
  LOGO_MARK: env("NEXT_PUBLIC_APP_LOGO_MARK", "Q"),
  LOGO_URL: env("NEXT_PUBLIC_APP_LOGO_URL", ""),
  SUPPORT_EMAIL: env("NEXT_PUBLIC_SUPPORT_EMAIL", "support@qguide.bio"),
  LEGAL_COMPANY_NAME: env("NEXT_PUBLIC_LEGAL_COMPANY_NAME", "QGuide"),
  WEBSITE_URL: env("NEXT_PUBLIC_APP_WEBSITE_URL", ""),
  DOCS_URL: env("NEXT_PUBLIC_APP_DOCS_URL", ""),
};

/** Lowercase, filesystem/URL-safe form of the app name (export filenames, ids). */
export const APP_SLUG = BRANDING.APP_NAME.toLowerCase().replace(/[^a-z0-9]+/g, "");

/** One-line research-use notice — keep in sync with `legal.SHORT_DISCLAIMER`. */
export const SHORT_DISCLAIMER =
  `${BRANDING.APP_NAME} output is a computational estimate for research use only — ` +
  `not validated laboratory or clinical advice. Validate experimentally.`;

export const {
  APP_NAME,
  APP_DESCRIPTION,
  TAGLINE,
  LOGO_MARK,
  SUPPORT_EMAIL,
  LEGAL_COMPANY_NAME,
} = BRANDING;

File: qguide/app/branding.py
"""
Central branding / product-identity configuration.

Every user-visible mention of the product name, tagline, support address or legal
entity should read from here (or from the mirrored frontend module
`web/lib/branding.ts`) rather than hardcoding a string. Renaming the product is
then a one-line change here plus the matching env vars -- no code search.

Every field is overridable by an environment variable so a rebrand (or a
white-label deployment) needs no code change at all.
"""
from __future__ import annotations

import os
from dataclasses import dataclass, asdict
from typing import Dict


def _env(key: str, default: str) -> str:
    v = os.environ.get(key)
    return v.strip() if v and v.strip() else default


@dataclass(frozen=True)
class Branding:
    app_name: str
    app_short_name: str
    app_description: str
    tagline: str
    logo_mark: str          # single-character/emoji mark used in compact chrome
    logo_url: str           # optional full logo asset (empty -> use logo_mark)
    support_email: str
    legal_company_name: str
    website_url: str
    docs_url: str

    def as_dict(self) -> Dict[str, str]:
        return asdict(self)


BRANDING = Branding(
    app_name=_env("APP_NAME", "QGuide"),
    app_short_name=_env("APP_SHORT_NAME", _env("APP_NAME", "QGuide")),
    app_description=_env(
        "APP_DESCRIPTION",
        "Context-aware, explainable, quantum-assisted CRISPR guide RNA design. "
        "Optimises for predicted biological outcome, not just cutting.",
    ),
    tagline=_env("APP_TAGLINE", "Explainable, outcome-first CRISPR guide design"),
    logo_mark=_env("APP_LOGO_MARK", "Q"),
    logo_url=_env("APP_LOGO_URL", ""),
    support_email=_env("SUPPORT_EMAIL", "support@qguide.bio"),
    legal_company_name=_env("LEGAL_COMPANY_NAME", "QGuide"),
    website_url=_env("APP_WEBSITE_URL", ""),
    docs_url=_env("APP_DOCS_URL", ""),
)

# Convenience aliases (import these rather than re-reading the env).
APP_NAME = BRANDING.app_name
APP_DESCRIPTION = BRANDING.app_description
TAGLINE = BRANDING.tagline
SUPPORT_EMAIL = BRANDING.support_email
LEGAL_COMPANY_NAME = BRANDING.legal_company_name
